### PR TITLE
Card API cleanup

### PR DESCRIPTION
**Pull request by misterzed88**
Improve card API design without breaking anything and without changing any functionality.

The card API (especially sc_pin_cmd_data) has become cluttered with fields with unclear purposes, or even with fields which are used only privately in some card drivers. Some fields have also become obsoleted by recent changes in master.

The existence of these fields makes it more complex than necessary to implement new card drivers, since it is not clear how the fields should be handled. The excess fields also make the API harder to use, since it is not obvious which of the fields can be relied upon to be available or not.

In short: this pull request is an attempt at cleaning up, removing unused- and private fields from the public interface and improving public comments.

##### Checklist

- [ ] Documentation is added or updated
- [ ] New files have a LGPL 2.1 license statement
- [X] PKCS#11 module is tested
- [ ] Windows minidriver is tested
- [ ] macOS tokend is tested

### CODE REVIEW

**Comment by Jakuje** [comment]
I have no idea how the OSX build passed for this PR as it looks like `length_offset` is still used in the `src/libopensc/reader-cryptotokenkit.m` and it is failing now:

https://travis-ci.org/github/OpenSC/OpenSC/jobs/720459495

I am not sure what it is used there for as I am not familiar with the osx code. Can you check if we should revert 48d939b0 or there is a way to get it working on OSX too?

**Comment by misterzed88** [comment]
Oops, I was not aware of the OSX Cryptotokenkit based reader module and it was therefore not kept in synch when making the PIN-pad changes in reader-pcsc.c. The PR test build does not complain since it builds without Cryptotokenkit support (and does therefore not build this module):
```CryptoTokenKit support:  no```

I created a fix in PR #2099.  This is the preferred way of fixing it (instead of reverting commit 48d939b05719e0e47af7891bf438c09647929f64). In this quick timeframe I have not had any chance to build and test on OSX.

In about 2 weeks timeframe I should be able to do a real test run on OSX (which would also include testing with PIN-pad readers).

**Comment by Jakuje** [comment]
Thank you for investigation and for a quick fix. As I do not know much about the OSX, I already asked Frank for review.
